Build failure in MSVC 2022

<!-- For information about contributing see: https://github.com/supercollider/supercollider/wiki/Contributing-directory -->
<!-- Please check there isn't already an issue on this topic. -->

## Environment

* SuperCollider version: `develop`
* Operating system: Windows
* Other details (Qt version, audio driver, etc.): MSVC 2022

## Steps to reproduce

```sh
cmake -G "Visual Studio 17 2022" -A "x64" -D CMAKE_PREFIX_PATH="$Qt5_DIR" -D SUPERNOVA=ON ..
cmake --build . --config Release --target install
```

## Expected vs. actual behavior

The build stops here:
<details>

```
 
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,17): error C2988: unrecognizable template declaration/definition [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,17): error C2143: syntax error: missing ')' before '(' [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,17): error C2146: syntax error: missing ')' before identifier '_Ch' [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,17): error C2065: '_Ch': undeclared identifier [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,17): error C2365: 'std::isdigit': redefinition; previous definition was 'function' [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,17): error C2059: syntax error: ')' [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,17): error C2059: syntax error: '||' [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,57): error C2143: syntax error: missing ';' before '{' [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
C:\Program Files\Microsoft Visual Studio\2022\Enterprise\VC\Tools\MSVC\14.41.34120\include\locale(221,57): error C2447: '{': missing function header (old-style formal list?) [D:\a\supercollider\supercollider\build\lang\libsclang.vcxproj]
  (compiling source file '../../lang/LangSource/SC_TerminalClient.cpp')
  
```
</details>

Can be observed in #6489 

Expected: the build finishes :)

<!-- Paste error messages in entirety. Use gist if very long. -->

--- a/lang/CMakeLists.txt
+++ b/lang/CMakeLists.txt
@@ -335,6 +335,10 @@ if (GC_SANITYCHECK)
 	target_compile_definitions(libsclang PUBLIC GC_SANITYCHECK)
 endif()
 
+if(MSVC)
+    target_compile_definitions(libsclang PUBLIC HAVE_ISXDIGIT) # needed for MSVC 2022
+endif()
+
 if(CMAKE_SYSTEM_NAME MATCHES "Linux")
 	target_link_libraries(libsclang rt)
 endif()

